pkg/test/framework/components/cluster/cluster.go

```diff
@@ -180,4 +192,8 @@ type Cluster interface {
 
 	// ConfigName returns the name of the config cluster for this cluster.
 	ConfigName() string
+
+	// Versions returns the set of Istio control plane versions running in this cluster
+	// returns nil if the cluster is not primary or there are no revisions running
+	Versions() Versions
```

Comment:
Hmm ... should we define a `ControlPlane` abstraction and have `Versions` be there? Then this starts to step on the `Istio` component a bit. Perhaps this shouldn't be an attribute on the cluster, but be accessible through `Istio` somehow?

@stevenctl WDYT?

Reply:
The `Istio` component could hold this, I don't see a reason to add another component. 

I think even non-primary clusters would have versions, since the proxies there would connect to various versioned control planes. 

If a requirement comes along to install some versions on a cluster, and other versions on another, then I can see this in the `Cluster`.

WRT config, we can change the topology config files we pass in so that they're not an array at the top level, to allow global properties to be specified.

Reply:
I think part of the problem might be that we've already dirtied the cluster abstraction with control plane information.  Fundamentally, a cluster is just a bucket for workloads and shouldn't know about the specific things that run there. Giving cluster a list of control plane information blurs the line of responsibility between the clusters and the Istio component. Istio is our control plane component and has to maintain that information anyway ... it would seem a better home for this sort of thing.

Also it's important to separate our configuration (i.e yaml) structure from our runtime structure. If we add these properties to "cluster" in the yaml, that matters much less. After it's parsed, however, that information should go to Istio and not appear on the runtime cluster object.

Reply:
Could make sense to stick the versions on `Environment`, and then pass through to the `Istio` component when needed?

We have access to the `Environment` component in places where we won't necessarily have Istio component access (like when deciding whether to skip tests based on labels like `MinIstioVersion` and echo deployment logic).

Reply:
Maybe our topology info should just be in the top-level settings object? That way, it's just attached to the context.